src/console/console_cmd.c

```
@@ -294,6 +349,92 @@ int console_cmd_rank(game_state *gs, int argc, char **argv) {
+    if(!har_obj) {
+        return 1;
+    }
+
+    // Parse LHS: harX.attr
+    char *lh_har_part = argv[1];
+    char *dot_pos = strchr(lh_har_part, '.');
+    if(dot_pos == NULL) {
+        console_output_addline("Invalid LHS format. Expected harX.attr");
+        return 1;
+    }
+
+    // Split into har and attribute parts
+    *dot_pos = '\0'; // Terminate har part
+    char *lh_attr_part = dot_pos + 1;
+
+    rec_assertion op;
+
+    int res = rec_assertion_get_operand(&op.operand1, lh_har_part, lh_attr_part);
+    if(res == 1) {
+        console_output_addline("Invalid LHS har identifier. Use har1 or har2.\n");
+        return 1;
+    } else if(res == 2) {
+        console_output_addline("Invalid LHS har attribute");
+    }
+
+    // Parse operator
+    const char *oper = argv[2];
+
+    if(strcmp(oper, ":=") == 0) {
+        op.op = OP_SET;
+    } else if(strcmp(oper, "==") == 0) {
+        op.op = OP_EQ;
+    } else if(strcmp(oper, ">") == 0) {
+        op.op = OP_GT;
+    } else if(strcmp(oper, "<") == 0) {
+        op.op = OP_LT;
+    } else {
+        console_output_addline("Invalid operator. Use ==, >, <, or :=.");
+        return 1;
+    }
+
+    char *rh_har_part = argv[3];
+    dot_pos = strchr(argv[3], '.');
+    if(dot_pos != NULL) { // RHS is har.attr
+        *dot_pos = '\0';  // Terminate har part
+        char *rh_attr_part = dot_pos + 1;
+
+        int res = rec_assertion_get_operand(&op.operand2, rh_har_part, rh_attr_part);
+        if(res == 1) {
+            console_output_addline("Invalid RHS har identifier. Use har1 or har2.\n");
+            return 1;
+        } else if(res == 2) {
+            console_output_addline("Invalid RHS har attribute");
+        }
+
+    } else { // RHS is integer
+        char *endptr;
+        long rhs_long = strtol(argv[3], &endptr, 10);
+        if(endptr == argv[3] || *endptr != '\0') {
+            console_output_addline("Invalid RHS integer literal.");
```

rec_assertion_get_operand will silently accept malformed integer literals like `literal.smiley`, interpreting them as 0.

---

Fixed.